pkg/specgen/specgen.go

```diff
@@ -467,7 +467,12 @@ type ContainerNetworkConfig struct {
 	// UseImageHosts indicates that /etc/hosts should not be managed by
 	// Podman, and instead sourced from the image.
 	// Conflicts with HostAdd.
-	UseImageHosts bool `json:"use_image_hosts,omitempty"`
+	// Do not set omitempty here, if this is false it should be set to not get
+	// the server default.
+	// Ideally this would be a pointer so we could differentiate between an
+	// explicitly false/true and unset (containers.conf default). However
+	// specgen is stable so we can not change this right now.
+	UseImageHosts bool `json:"use_image_hosts"`
```

Comment:
Would changing to a pointer be a breaking change in the generated JSON? I think that's where our stability guarantee is, not on the structure itself.

Reply:
But the structure is used in the bindings.

I think we have to start documentation what is stable and what not. The bindings import other places so we have to keep track of that and cannot break potential users.

Reply:
This feels like a good cabal topic.

Reply:
After discussion, we can't break the bindings, but we should make sure this is marked so we can actually fix it in Podman 5